packages/bundle-size/src/reporters/markdownReporter.js

```diff
@@ -0,0 +1,151 @@
+    if (reportEntry.diff.gzip.delta === 0 || reportEntry.diff.minified.delta === 0) {
+      unchangedEntries.push(reportEntry);
+    } else {
+      changedEntries.push(reportEntry);
+    }
+  });
+
+  return {
+    changedEntries: sortComparedReport(changedEntries),
+    unchangedEntries: sortComparedReport(unchangedEntries),
+  };
+}
+
+/**
+ * @param {import('../utils/compareResultsInReports').ComparedReport} result
+ * @param {string} commitSHA
+ * @param {boolean} quiet
+ */
+module.exports = async function markdownReporter(result, commitSHA, quiet) {
+  const packageRoot = findPackageRoot(__dirname);
+
+  if (!packageRoot) {
+    throw new Error(
+      [
+        'Failed to find a package root (directory that contains "package.json" file)',
+        `Lookup start in: ${__dirname}`,
+      ].join('\n'),
+    );
+  }
+
+  const artifactsDir = path.resolve(packageRoot, 'dist');
+  const artifactsFilename = path.join(artifactsDir, 'bundle-size.md');
+
+  const report = [];
+
+  report.push('## 📊 Bundle size report');
+  report.push('');
+
+  const { changedEntries, unchangedEntries } = partitionReport(result);
+
+  if (changedEntries.length > 0) {
+    report.push('| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |');
+    report.push('| :---------------- | -----------------------: | ----: | ---------: |');
+
+    changedEntries.forEach(entry => {
+      const title = `<samp>${entry.packageName}</samp> <br /> <abbr title='${entry.path}'>${entry.name}</abbr>`;
+      const before = entry.diff.empty
+        ? [`\`${formatBytes(0)}\``, '<br />', `\`${formatBytes(0)}\``].join('')
+        : [
+            `\`${formatBytes(entry.minifiedSize + entry.diff.minified.delta)}\``,
+            '<br />',
+            `\`${formatBytes(entry.gzippedSize + entry.diff.gzip.delta)}\``,
+          ].join('');
+      const after = [`\`${formatBytes(entry.minifiedSize)}\``, '<br />', `\`${formatBytes(entry.gzippedSize)}\``].join(
+        '',
+      );
+      const difference = entry.diff.empty
+        ? '🆕 New entry'
+        : [`${formatDelta(entry.diff.minified)}`, '<br />', `${formatDelta(entry.diff.gzip)}`].join('');
+
+      report.push(`| ${title} | ${before} | ${after} | ${difference}|`);
+    });
+
+    report.push('');
+  }
+
+  if (unchangedEntries.length > 0) {
+    report.push('<details>');
+    report.push('<summary>Unchanged files</summary>');
+    report.push('');
+
+    report.push('| Package & Exports | Size (minified/GZIP) |');
+    report.push('| ----------------- | -------------------: |');
+
+    unchangedEntries.forEach(entry => {
+      const title = `<samp>${entry.packageName}</samp> <br /> <abbr title='${entry.path}'>${entry.name}</abbr>`;
+      const size = [`\`${formatBytes(entry.minifiedSize)}\``, '<br />', `\`${formatBytes(entry.gzippedSize)}\``].join(
+        '',
+      );
+
+      report.push(`| ${title} | ${size} |`);
+    });
+
+    report.push('</details>');
+  }
+
+  report.push(
```

Comment:
the amount of these `report.push` calls with string interpolation looks kinda scary, have you considered using an ejs template ?

Reply:
Agree 😓

As I know EJS is designed for HTML templates and is not supported by Prettier (https://github.com/prettier/prettier/issues/8676#issuecomment-651756744). I tried to rewrite this to EJS template and I don't think that it become more readable 😕

Do you think that we should go with ejs? Are there any other options?

Reply:
ejs might be good idea, but we lost all TS advantages....

I'd keep it within js for now, although it would be nice to make it more readable.

Reply:
I quickly hacked and came up with something like this 

```js
async function markdownReporter(result, commitSHA, quiet) {
  const packageRoot = findPackageRoot(__dirname);

  if (!packageRoot) {
    throw new Error(
      [
        'Failed to find a package root (directory that contains "package.json" file)',
        `Lookup start in: ${__dirname}`,
      ].join('\n'),
    );
  }

  const artifactsDir = path.resolve(packageRoot, 'dist');
  const artifactsFilename = path.join(artifactsDir, 'bundle-size.md');

  const { changedEntries, unchangedEntries } = getChangedEntriesInReport(result);

  const template = stripIndents`
  ## 📊 Bundle size report
  ${printChanged(changedEntries)}
  ${printUnchanged(unchangedEntries)}

  <sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/${commitSHA}'>${commitSHA}</a></sub>
  `;

  await fs.mkdir(artifactsDir, { recursive: true });
  await fs.writeFile(artifactsFilename, prettier.format(template, { parser: 'markdown' }));

  if (!quiet) {
    console.log([chalk.blue('[i]'), `A report file was written to ${artifactsFilename}`].join(' '));
  }
};

/**
 *
 * @param {getChangedEntriesInReport.ComparedReport} entries
 */
function printChanged(entries) {
  if (entries.length > 0) {
    const template = `
  | Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
  | :---------------- | -----------------------: | ----: | ---------: |
  ${entries.map(newChangedEntry)}

  `;

    return template;
  }

  return '';
}

/**
 *
 * @param {getChangedEntriesInReport.ComparedReport} entries
 */
function printUnchanged(entries) {
  if (entries.length > 0) {
    const template = `
<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
${entries.map(newUnchangedEntry)}
</details>
 `;

    return template;
  }

  return '';
}

/**
 *
 * @param {getChangedEntriesInReport.ComparedReport[number]} entry
 */
function newUnchangedEntry(entry) {
  const title = `<samp>${entry.packageName}</samp> <br /> <abbr title='${entry.path}'>${entry.name}</abbr>`;
  const size = [`\`${formatBytes(entry.minifiedSize)}\``, '<br />', `\`${formatBytes(entry.gzippedSize)}\``].join('');
  return `| ${title} | ${size} |`;
}

/**
 *
 * @param {getChangedEntriesInReport.ComparedReport[number]} entry
 */
function newChangedEntry(entry) {
  const title = `<samp>${entry.packageName}</samp> <br /> <abbr title='${entry.path}'>${entry.name}</abbr>`;
  const before = entry.diff.empty
    ? [`\`${formatBytes(0)}\``, '<br />', `\`${formatBytes(0)}\``].join('')
    : [
        `\`${formatBytes(entry.minifiedSize + entry.diff.minified.delta)}\``,
        '<br />',
        `\`${formatBytes(entry.gzippedSize + entry.diff.gzip.delta)}\``,
      ].join('');
  const after = [`\`${formatBytes(entry.minifiedSize)}\``, '<br />', `\`${formatBytes(entry.gzippedSize)}\``].join('');
  const difference = entry.diff.empty
    ? '🆕 New entry'
    : [`${formatDelta(entry.diff.minified)}`, '<br />', `${formatDelta(entry.diff.gzip)}`].join('');

  return `| ${title} | ${before} | ${after} | ${difference}|`;
}

```

Reply:
I am not sure that it improves something drastically, but I can apply it if you want ¯\_(ツ)_/¯

Reply:
if ejs does not seem to improve smth then we can keep it like this. My thought was that it would remove the need for all the escaping and you could just pass a data object to tokenize the content